dsls/sql/src/main/java/org/apache/beam/dsls/sql/transform/BeamAggregationTransform.java

```diff
@@ -117,4 +132,540 @@ public Instant apply(BeamSQLRow input) {
     }
   }
 
+  /**
+   * Aggregation function which supports COUNT, MAX, MIN, SUM, AVG.
+   *
+   * <p>Multiple aggregation functions are combined together.
+   * For each aggregation function, it may accept part of all data types:<br>
+   *   1). COUNT works for any data type;<br>
+   *   2). MAX/MIN works for INT, LONG, FLOAT, DOUBLE, DECIMAL, SMALLINT, TINYINT, TIMESTAMP;<br>
+   *   3). SUM/AVG works for INT, LONG, FLOAT, DOUBLE, DECIMAL, SMALLINT, TINYINT;<br>
+   *
+   */
+  public static class AggregationCombineFn extends CombineFn<BeamSQLRow, BeamSQLRow, BeamSQLRow> {
+    private BeamSQLRecordType aggDataType;
+
+    private int countIndex = -1;
+
+    List<String> aggFunctions;
+    List<BeamSqlExpression> aggElementExpressions;
+
+    public AggregationCombineFn(List<AggregateCall> aggregationCalls,
+        BeamSQLRecordType sourceRowRecordType) {
+      this.aggDataType = new BeamSQLRecordType();
+      this.aggFunctions = new ArrayList<>();
+      this.aggElementExpressions = new ArrayList<>();
+
+      boolean hasAvg = false;
+      boolean hasCount = false;
+      int countIndex = -1;
+      for (int idx = 0; idx < aggregationCalls.size(); ++idx) {
+        AggregateCall ac = aggregationCalls.get(idx);
+        //verify it's supported.
+        verifySupportedAggregation(ac);
+
+        aggDataType.addField(ac.name, ac.type.getSqlTypeName());
+
+        SqlAggFunction aggFn = ac.getAggregation();
+        switch (aggFn.getName()) {
+        case "COUNT":
+          aggElementExpressions.add(BeamSqlPrimitive.<Long>of(SqlTypeName.BIGINT, 1L));
+          hasCount = true;
+          countIndex = idx;
+          break;
+        case "SUM":
+        case "MAX":
+        case "MIN":
+        case "AVG":
+          int refIndex = ac.getArgList().get(0);
+          aggElementExpressions.add(new BeamSqlInputRefExpression(
```

Comment:
It assumes the param of `SUM` is a table column here. But actually not, the following is a valid sql:

```sql
select sum(1) from t1
```